mobile/src/main/java/rs/readahead/washington/mobile/views/activity/VideoViewerActivity.kt

```diff
@@ -0,0 +1,569 @@
+        //TODO CHECK ERROR MSG WHEN RENAME
+        DialogUtils.showBottomMessage(
+            this,
+            getString(R.string.gallery_toast_fail_deleting_files),
+            true
+        )
+    }
+
+    override fun getContext(): Context {
+        return this
+    }
+
+    override fun dispatchKeyEvent(event: KeyEvent): Boolean {
+        // Show the controls on any key event.
+        simpleExoPlayerView.showController()
+        // If the event was not handled then see if the player view can handle it as a media key event.
+        return super.dispatchKeyEvent(event) || simpleExoPlayerView.dispatchMediaKeyEvent(event)
+    }
+
+    private fun shareMediaFile() {
+        if (vaultFile == null) {
+            return
+        }
+        if (vaultFile!!.metadata != null) {
+            showShareWithMetadataDialog()
+        } else {
+            startShareActivity(false)
+        }
+    }
+
+    private fun startShareActivity(includeMetadata: Boolean) {
+        if (vaultFile == null) {
+            return
+        }
+        MediaFileHandler.startShareActivity(this, vaultFile, includeMetadata)
+    }
+
+    private fun initializePlayer() {
+        val needNewPlayer = player == null
+        if (needNewPlayer) {
+
+            player = ExoPlayer.Builder(this).build()
+            player?.let { exo ->
+                exo.playWhenReady = true
+                binding.playerView.player = exo
+            }
+
+            simpleExoPlayerView.setPlayer(player)
+            player!!.playWhenReady = shouldAutoPlay
+        }
+        if (needNewPlayer || needRetrySource) {
+            if (intent.hasExtra(VIEW_VIDEO) && intent.extras != null) {
+                val vaultFile = intent.extras!![VIEW_VIDEO] as VaultFile?
+                if (vaultFile != null) {
+                    this.vaultFile = vaultFile
+                    toolbar!!.title = vaultFile.name
+                    setupMetadataMenuItem(vaultFile.metadata != null)
+                }
+            }
+            val mediaFileDataSourceFactory = MediaFileDataSourceFactory(
+                this,
+                vaultFile!!, null
+            )
+
+            val mediaItem = MediaItem.fromUri(MediaFileHandler.getEncryptedUri(this, vaultFile))
+            val mediaSource: MediaSource = ProgressiveMediaSource.Factory(mediaFileDataSourceFactory)
+                .createMediaSource(mediaItem)
+
+            val haveResumePosition = resumeWindow != C.INDEX_UNSET
+            if (haveResumePosition) {
+                player!!.seekTo(resumeWindow, resumePosition)
+            }
+            player!!.prepare(mediaSource, !haveResumePosition, false)
+            needRetrySource = false
+        }
+    }
+
+    private fun releasePlayer() {
+        if (player != null) {
+            shouldAutoPlay = player!!.playWhenReady
+            //updateResumePosition(); // todo: fix source skipping..
+            player!!.release()
```

Comment:
same here we already checked that player is not null, let's apply the same logic on the rest of the class.